### ISSUE TITLE
yield from is not allowed in async function

### ISSUE THREAD

**Issue by wyuenho** [MartinThoma/flake8-simplify#146]
## Desired change
Do not warning if a SIM104 is encountered inside an async function or method.

* **Rule(s)**: SIM104
* **Adjustment**: In which way should it/they be adjusted?
Do not warning if a SIM104 is encountered inside an async function or method.

## Explanation

Why is the adjustment necessary / better?

The suggested fix to replace a for yield loop with a `yield from` statement is [invalid](https://peps.python.org/pep-0525/#asynchronous-yield-from) inside an async function.

## Example

This is an example where the mentioned rule(s) would currently be suboptimal:

```python
async fib(limit=10):
    if limit == 1:
        yield 0
    if limit == 2:
        yield 0
        yield 1
    i, j = 0, 1
    for _ in range(2, limit):
        n = i + j
        yield n
        i, j = j, n
```

**Comment by MartinThoma** [comment]
Looks like a duplicate of https://github.com/MartinThoma/flake8-simplify/issues/67 . Which version of flake8-simplify do you use?

**Comment by wyuenho** [comment]
I'm on 0.19.2

**Comment by wyuenho** [comment]
Looks like the `ast.For` node doesn't have a parent attribute and you've ignore type checking on that line so you didn't catch this? How does flake8 handle exceptions throw from a rule?
